### PR TITLE
Adding oracle.can_connect metric

### PR DESCRIPTION
**Pull request by nenadnoveljic**
### What does this PR do?

Adding `oracle.can_connect` check.

### Motivation

The metric is a prerequisite for the service check.

### QA Testing

Check the metric [here](https://dd.datad0g.com/check/summary?id=oracle.can_connect). Restart the database and check if the status is changing.

### CODE REVIEW

File: pkg/collector/corechecks/oracle-dbm/oracle.go [hunk 1]
```diff
@@ -78,24 +79,58 @@ type Check struct {
...
+	}
+
+	message := ""
+	var status servicecheck.ServiceCheckStatus
+	if err == nil {
+		status = servicecheck.ServiceCheckOK
+	} else {
+		status = servicecheck.ServiceCheckCritical
+		message = fmt.Sprintf("%s", err)
+		log.Errorf("connect error %s", message)
```
Comment (alexandre-normand) [inline pkg/collector/corechecks/oracle-dbm/oracle.go:95]: Couldn't this just be this?
```suggestion
		log.Errorf("failed to connect: %s", err)
```

File: pkg/collector/corechecks/oracle-dbm/statements.go [hunk 2]
```diff
@@ -607,29 +607,33 @@ func (c *Check) StatementMetrics() (int, error) {
...
+					}
+					log.Tracef("Query metrics fqt payload %s", string(FQTPayloadBytes))
+					sender.EventPlatformEvent(FQTPayloadBytes, "dbm-samples")
+					c.fqtEmitted.Set(queryRow.QuerySignature, "1", cache.DefaultExpiration)
 				}
-				log.Tracef("Query metrics fqt payload %s", string(FQTPayloadBytes))
-				sender.EventPlatformEvent(FQTPayloadBytes, "dbm-samples")
-				c.fqtEmitted.Set(queryRow.QuerySignature, "1", cache.DefaultExpiration)
+			} else {
+				log.Error("fqtEmitted = nil")
```
Comment (alexandre-normand) [inline pkg/collector/corechecks/oracle-dbm/statements.go:636]: When would `fqtEmitted` be nil? Would this be when the check is currently shutting down? If so, we might want to just include a note about that so it's more obvious to users/support how to interpret this error log.

**Review by alai97 — APPROVED** [review]
Looks good for docs!

File: pkg/collector/corechecks/oracle-dbm/statements.go (hunk 2 above)
Comment (nenadnoveljic) [reply pkg/collector/corechecks/oracle-dbm/statements.go]: Yes the check was shutting down because the database restarted. I enhanced the error message.

**Review by alexandre-normand — APPROVED** [review]

**Pull request merged**